### PR TITLE
Am/c api/multi bit keygen

### PR DESCRIPTION
**Pull request by IceTDrinker**
@agnesLeroy no C header as discussed with @tmontaigu you will need to incorporate a build step for the C API for TFHE-rs, if you need infos on feature configs let us know !

### CODE REVIEW

File: tfhe/src/c_api/core_crypto/mod.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            output_glwe_sk_dim.to_glwe_size(),
+            output_glwe_sk_poly_size,
+            lwe_multi_bit_base_log,
+            lwe_multi_bit_level_count,
+            input_lwe_sk_dim,
+            lwe_multi_bit_grouping_factor,
+            CiphertextModulus::new_native(),
+        );
+
+        *result = bsk.into_container().len();
```
Comment (IceTDrinker) [inline tfhe/src/c_api/core_crypto/mod.rs:39]: can definitely avoid an allocation here, it was done quick and dirty (yikes)

Comment (agnesLeroy) [reply tfhe/src/c_api/core_crypto/mod.rs:39]: multi-bit pbs key has this size:

```
int multi_bit_bsk_size = lwe_dimension * pbs_level * (glwe_dimension + 1) *
                 (glwe_dimension + 1) * polynomial_size *
                 (1 << grouping_factor) / grouping_factor;
```

**Review by agnesLeroy — APPROVED** [review]
I've written the multi-bit bsk size in comment in case you want to remove the unnecessary alloc, otherwise looks good to me. Thanks a lot!

File: tfhe/src/c_api/core_crypto/mod.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/src/c_api/core_crypto/mod.rs:39]: yeah, I'll re-use some primitives from concrete-core

Comment (IceTDrinker) [reply tfhe/src/c_api/core_crypto/mod.rs]: @agnesLeroy be careful there is an API break, if you can just check the new entry point works ok for you ?

Comment (IceTDrinker) [reply tfhe/src/c_api/core_crypto/mod.rs]: the formula looks good on your end anyways, but you can use the entry point from the C API to have consistency if you want

**Pull request merged**